homeassistant/components/mqtt/__init__.py

```diff
@@ -432,23 +424,29 @@ def async_publish_service(call):
     return True
 
 
+Subscription = namedtuple('Subscription',
+                          ['topic', 'callback', 'qos', 'encoding'])
+Subscription.__new__.__defaults__ = (0, 'utf-8')
+
+Message = namedtuple('Message', ['topic', 'payload', 'qos', 'retain'])
+Message.__new__.__defaults__ = (0, False)
```

Comment:
While we're at creating `namedtuple`s, let's also create another struct to copy MQTT messages around.